sql/core/src/test/scala/org/apache/spark/sql/JoinSuite.scala

```diff
@@ -1147,4 +1147,40 @@ class JoinSuite extends QueryTest with SharedSparkSession with AdaptiveSparkPlan
       })
     }
   }
+
+  test("SPARK-32290: SingleColumn Null Aware Anti Join Optimize") {
+    withSQLConf(SQLConf.NULL_AWARE_ANTI_JOIN_OPTIMIZE_ENABLED.key -> "true",
+      SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> Long.MaxValue.toString) {
```

Comment:
do we need to set this config?

Reply:
will remove it.

Reply:
kind of need broadcastThreshold config
or the following case will planed into SMJ

```
// negative hand-written left anti join
      // testData.key nullable false
      // testData2.a nullable false
      joinExec = assertJoin((
        "select * from testData left anti join testData2 ON key = a or isnull(key = a)",
        classOf[BroadcastHashJoinExec]))
      assert(!joinExec.asInstanceOf[BroadcastHashJoinExec].isNullAwareAntiJoin)
```